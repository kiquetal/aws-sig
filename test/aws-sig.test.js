"use strict";

const config = require("./config.js");

const sign = require("../dist/aws-sig.cjs.js");

describe("aws-sig", () => {
    it("Should sort SignedHeaders by name", () => {
        expect(
            sign({
                url     : "https://aws.amazon.com",
                headers : {
                    "X-Amz-Date" : "20150830T123600Z",

                    z : 1,
                    a : 1,
                    m : 1
                }
            }, config())
        ).toMatchSnapshot();
    });
});
