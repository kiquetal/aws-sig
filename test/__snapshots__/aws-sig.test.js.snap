// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aws-sig should be precise about the missing config param values 1`] = `"Missing required config fields: accessKeyId, region"`;

exports[`aws-sig should sort SignedHeaders by name 1`] = `
Object {
  "headers": Object {
    "Authorization": "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=a;m;x-amz-date;z, Signature=d5b729b777af569c5e22fbf1d1a27fad8d490b6fc3c7a5e1a54542bfa68c0998",
    "X-Amz-Date": "20150830T123600Z",
    "a": 1,
    "m": 1,
    "z": 1,
  },
  "test": Object {
    "auth": "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=a;m;x-amz-date;z, Signature=d5b729b777af569c5e22fbf1d1a27fad8d490b6fc3c7a5e1a54542bfa68c0998",
    "canonical": "GET
/

a:1
m:1
x-amz-date:20150830T123600Z
z:1

a;m;x-amz-date;z
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
    "sts": "AWS4-HMAC-SHA256
20150830T123600Z
20150830/us-east-1/service/aws4_request
3d94ead6a8f2253deb74d7183e308aaa8c311c7f8248b19b69667f65639de28c",
  },
  "url": "https://aws.amazon.com",
}
`;

exports[`aws-sig should throw on a missing config param values 1`] = `"Missing required config fields: accessKeyId, region, secretAccessKey, service"`;

exports[`aws-sig should throw on a missing request param 1`] = `"Missing request object"`;

exports[`aws-sig should throw on missing config param 1`] = `"Missing config object"`;
